schemas/v3.schema.graphql

```diff
@@ -0,0 +1,764 @@
+  tokens: [PriceOracleAsset!]! @derivedFrom(field: "oracle")
+  usdPriceEthHistory: [UsdEthPriceHistoryItem!]! @derivedFrom(field: "oracle")
+  tokensWithFallback: [PriceOracleAsset!]!
+  lastUpdateTimestamp: Int!
+  version: Int!
+  baseCurrency: Bytes! # new for v3
+  baseCurrencyUnit: BigInt! # new for v3
+}
+
+# Entity representing A/VariableDebt/StableDebt tokens
+type SubToken @entity {
+  """
+  SubToken address
+  """
+  id: ID!
+  pool: Pool!
+  tokenContractImpl: Bytes
+  underlyingAssetAddress: Bytes!
+  underlyingAssetDecimals: Int!
+  rewards: [Rewards!]! @derivedFrom(field: "asset")
+}
+
+type Referrer @entity {
+  id: ID!
+  supplies: [Supply!]! @derivedFrom(field: "referrer")
+  borrows: [Borrow!]! @derivedFrom(field: "referrer")
+}
+
+interface UserTransaction {
+  id: ID!
+  pool: Pool!
+  user: User!
+  timestamp: Int!
+}
+
+type Supply implements UserTransaction @entity {
+  """
+  tx hash
+  """
+  id: ID!
+  pool: Pool!
+  user: User!
+  caller: User!
+  reserve: Reserve!
+  referrer: Referrer
+  userReserve: UserReserve!
+  amount: BigInt!
+  timestamp: Int!
+}
+
+type RedeemUnderlying implements UserTransaction @entity {
+  """
+  tx hash
+  """
+  id: ID!
+  pool: Pool!
+  user: User!
+  to: User!
+  reserve: Reserve!
+  userReserve: UserReserve!
+  amount: BigInt!
+  timestamp: Int!
+}
+
+type Borrow implements UserTransaction @entity {
+  """
+  tx hash
+  """
+  id: ID!
+  pool: Pool!
+  user: User!
+  caller: User!
+  reserve: Reserve!
+  userReserve: UserReserve!
+  amount: BigInt!
+  borrowRate: BigInt!
+  borrowRateMode: Int!
+  referrer: Referrer
+  timestamp: Int!
+  stableTokenDebt: BigInt!
+  variableTokenDebt: BigInt!
+}
+
+type SwapBorrowRate implements UserTransaction @entity {
+  """
+  tx hash
+  """
+  id: ID!
+  pool: Pool!
+  user: User!
+  reserve: Reserve!
+  userReserve: UserReserve!
+  borrowRateModeFrom: Int!
+  borrowRateModeTo: Int!
+  stableBorrowRate: BigInt!
+  variableBorrowRate: BigInt!
+  timestamp: Int!
+}
+
+type UsageAsCollateral implements UserTransaction @entity {
+  """
+  tx hash
+  """ #
+  id: ID!
+  pool: Pool!
+  user: User!
+  reserve: Reserve!
+  userReserve: UserReserve!
+  fromState: Boolean!
+  toState: Boolean!
+  timestamp: Int!
+}
+
+type RebalanceStableBorrowRate implements UserTransaction @entity {
+  """
+  tx hash
+  """ #
+  id: ID!
+  pool: Pool!
+  user: User!
+  reserve: Reserve!
+  userReserve: UserReserve!
+  borrowRateFrom: BigInt!
+  borrowRateTo: BigInt!
+  #accruedBorrowInterest: BigInt!
+  timestamp: Int!
+}
+
+type Repay implements UserTransaction @entity {
+  """
+  tx hash
+  """
+  id: ID!
+  pool: Pool!
+  user: User!
+  repayer: User!
+  reserve: Reserve!
+  userReserve: UserReserve!
+  amount: BigInt!
+  timestamp: Int!
+  useATokens: Boolean!
+}
+
+type IsolationModeTotalDebtUpdated @entity {
+  """
+  tx hash
+  """
+  id: ID!
+  pool: Pool!
+  reserve: Reserve!
+  isolatedDebt: BigInt!
+  timestamp: Int!
+}
+
+type FlashLoan @entity {
```

Comment:
Shouldn't this implement UserTransaction as well? 

Noticed this in V2 schemas as well

Reply:
It doesnt apply since Flashloan doesn't store the user